### ISSUE TITLE
Add a Connector for Sybase Catalog in PrestoDB

### ISSUE THREAD

**Issue by nikhilsinghkushwah** [prestodb/presto#23633]
### Feature Request

**Feature Description:**
A connector for Sybase (SAP ASE) database in PrestoDB to enable querying data from Sybase catalogs directly using Presto.

**Use Case:**
Sybase is widely used in many legacy and enterprise applications. Adding a connector would allow users to query data stored in Sybase using Presto, benefiting organizations with mixed database environments, and providing seamless integration for data analysis alongside other supported databases (e.g., MySQL, PostgreSQL, Oracle).

**Proposed Solution:**
The connector would utilize the Sybase JDBC driver to interact with the database. It would support basic querying, metadata retrieval (schemas, tables, columns), and split management for parallel query execution. Key components would include:
- Sybase-specific metadata retrieval
- Efficient data fetching using JDBC
- Support for split management and parallelism based on table partitions or row ranges

**Potential Challenges:**
- Handling Sybase-specific data types and proprietary SQL extensions
- Performance tuning for large-scale queries
- Sybase's concurrency and transaction isolation mechanisms

**Alternatives:**
- Users can currently use JDBC-based tools to query Sybase but lack the Presto advantage of federated querying.
- Implementing a custom connector locally is an option, but a standardized official connector would be more widely beneficial.

**Additional Context:**
The Presto community would greatly benefit from the inclusion of this connector to enhance the platform’s ability to work with Sybase. We would be happy to contribute to its development or collaborate with maintainers to bring this feature to fruition.

**Related Issues or Discussions:**
N/A

**Comment by tdcmeehan** [comment]
@nikhilsinghkushwah would it be possible to include end to end tests with the introduction of this connector?

**Comment by elharo** [comment]
I agree this would be useful. Sybase is important in many organizations.

Does this have to be in the core repo for some reason or can it be published as an independent project? The latter could be much faster since it wouldn't require constant code reviews and approvals from Presto maintainers.

**Comment by nikhilsinghkushwah** [comment]
> @nikhilsinghkushwah would it be possible to include end to end tests with the introduction of this connector?

@tdcmeehan i have added standard test cases from connector perspective. Let me know if any other details are required.

**Comment by tdcmeehan** [comment]
@nikhilsinghkushwah If we want to include this in the main repo, we'll need to include end to end tests that run against some version of Sybase.  Unit test coverage of the various components won't be enough, we'll need integration style tests that prove query execution works end to end.  Are you able to add such tests?

**Comment by nikhilsinghkushwah** [comment]
@tdcmeehan Added integration test for end to end connector functionality.
